### PR TITLE
✨ use github-client internal rate-limits timer

### PR DESCRIPTION
**Pull request by slntopp**
Github Client is handling rate-limits internally out of the box, which means if rate limits are reached, client will return fake 403 API Rate limits response before even trying out retryable client.
Unfortunately we can't use `bypassRateLimitCheck` strategy, because it's private, but we can use `SleepUntilPrimaryRateLimitResetWhenRateLimited` instead, which should be doing just the same job as the retryable client.
Secondary limits must still be handled by retryable client.

### CODE REVIEW

File: providers/github/connection/connection.go [hunk 1]
```diff
@@ -67,8 +68,10 @@ func NewGithubConnection(id uint32, asset *inventory.Asset) (*GithubConnection,
 		}
 	}
 
+	ctx := context.WithValue(context.Background(), github.SleepUntilPrimaryRateLimitResetWhenRateLimited, true)
```
Comment (chris-rock) [inline providers/github/connection/connection.go:73]: Can we add a comment with your research findings, so that this context is written down in code?

**Review by chris-rock — COMMENTED** [review]
Looks great @slntopp Lets add some comments and we are good to go.

File: providers/github/connection/connection.go (hunk 1 above)
Comment (slntopp) [reply providers/github/connection/connection.go:73]: Added the comment

**Review by chris-rock — APPROVED** [review]
Thank you @slntopp

**Pull request merged**